sorting-algorithms.md created

denesnori opened the pull request:
New README on sorting algorithms including bubble sort, selection sort, insertion sort, quicksort and merge sort. Content: steps of the algorithm and its javascript implementation.

alwinsol committed:
Added JavaScript-Classes.md

Commit:
sorting-algorithms.md created

sofer commented:
@denesnori I love this. I wonder if it might be worth adding a reference to an explanation of the Big 0 notation used in your comparison table. Also, it might be nice to see if you can unearth anything about which algorithms are used in different implementations of JavaScript's Array.sort.

denesnori commented:
@sofer I found a Stackoverflow thread discussing the implementation of JS array.sort(). However, I have not yet figured out how I can wisely summarize it. http://stackoverflow.com/questions/234683/javascript-array-sort-implementation

I googled the Big-O notation, and wrote a few sections on it. But, it is quite long now and I don't use sorting algorithms as examples in my explanation, so I think is better to post it as a new readme.

sofer commented:
> I googled the Big-O notation, and wrote a few sections on it. But, it is quite long now and I don't use sorting algorithms as examples in my explanation, so I think is better to post it as a new readme.

@denesnori I agree. Why not start  a README on Big O and post a request in gitter for help with it.

alwinsol committed:
Updated JavaScript Classes Document

danmarcus111 commented on programmer-skills/sorting-algorithms.md:
```diff
@@ -0,0 +1,277 @@
+}
+
+
+function quickSort(array,left,right){
```
Why is an input of left and right required?  Presumably if you are sorting the whole array, left will be 0 and right will be array.length -1?

I understand that this allows you to select a portion of the array to sort, but as this isn't an aspect unique to quicksort perhaps it would be better to remove this functionality in the tutorial to make it cleaner to follow?

danmarcus111 commented:
Thanks for this @denesnori - I found this a very interesting read.  

Please ignore my earlier comment if you already read it.  As I went through the process in more detail I saw my error!

lucymonie committed:
New intro sorting-algorithms.md

lucymonie committed:
Small edits through entire file

lucymonie commented:
Hey Nora, I have added the intro as discussed. As I was reading through all the different approaches, I was thinking about exercises that people could do to help get clearer on the ideas. I recalled that Eloquent JavaScript has a reverse and reverse in place exercise in chapter 4 that might help people get a sense of the 'in place' idea. [Here's a link](http://eloquentjavascript.net/04_data.html#h_6xTmjj4Rf5) in case it could be helpful.

Cleop committed:
Proofreading changes

Great piece, I certainly learnt a few things! I've renamed the variable House as house as it's recommended to name variables beginning with a lower case letter (http://www.w3schools.com/js/js_conventions.asp). I've also rephrased a couple of bits slightly just to help them read a little clearer and changed some spelling/ grammatical errors. I've also introduced some syntax highlighting using markdowns.  Let me know if anything doesn't make sense!

Cleop committed:
2nd attempt!

Now with House objects as they correctly were before with a capital C! Looks ready to merge in my opinion.

Commit:
Fixed typos and tweaked explanations in a few places

alwinsol committed:
Update JavaScript-Classes.md

Incorporated latest comments

lucymonie committed:
Merge pull request #200 from codingforeveryone/JavaScript-updates

Added JavaScript-Classes.md

msachi commented:
Hey @denesnori , a few comments from my side as well:
- It's great that we have a README on the big O notation now, but it would still be useful to explain the concept in 1-2 sentences here, before you introduce the table
- I find the gif that you included for the Quick sort algorithm quite confusing (and too fast). It might be better to include the picture from algolist, for the sake of clarity and consistency
- Were you going to include the heap sort as well? If not you can list is at the end as an option for elaborating on your README

I will go on and fix a few typos / stylistic errors, hope that's ok!

msachi committed:
Typos / stylistic fixes

denesnori commented:
Hi!

Thanks for you comments. I agree with your comments. I will write about heap sort, find a picture explaining quicksort, and write a few words about the big-o notation. (Hopefully within a few days :))
I would be grateful if you fix typos and stylistic errors :),

msachi commented:
Cool, I did the fixes :)

denesnori committed:
Merge branch 'READMEs/add/sorting-algorithms' of https://github.com/codingforeveryone/READMEs into READMEs/add/sorting-algorithms

denesnori committed:
new section on the heapsort algorithm and minor fixes everywhere

denesnori commented:
Hi, I added a new section on heapsort, added a new image explaining quicksort, a sentence about the big-O notation (after that sending the readers to the big-O readme).
Hope you'll like it :)

msachi committed:
Update sorting-algorithms.md

Fixing typos

msachi commented on programmer-skills/sorting-algorithms.md:
```diff
@@ -0,0 +1,442 @@
+![Random array to heap](http://www.stoimen.com/blog/wp-content/uploads/2012/08/7.-Random-Array-to-Heap.png)   
+*Fig 9. is taken from [here](http://www.stoimen.com/blog/2012/08/07/computer-algorithms-heap-and-heapsort-data-structure/)*   
+
+####Heapify
```
If I understand this correctly, the heapify needs to be done several times recursively, i.e. once you've swapped you start from `Math.floor( array.length / 2)` again? If that's the case it might be worth spelling it out?

msachi commented on programmer-skills/sorting-algorithms.md:
```diff
@@ -0,0 +1,442 @@
+At start, we have no large elements at the end, and the heap at the front occupies all space. We will denote the size of the heap with heapsize,
+this is array.length at the beginning, and we will decrement it by one at each iteration step.
+
+At any given iteration, the largest element remaining in the heap is its root, physically stored as the first element in the array. The array(length-heapsize-1)
```
I'm not sure I understand this syntax (`array(length-heapsize-1)`), can you double check it? Did you mean `array.length-heapsize-1`? Or `array[array.length-heapsize-1]`?

msachi commented:
Hi @denesnori, nice additions. I added two comments for you to have a look, and fixed some typos myself.
